### ISSUE TITLE
Implement AcceptOffers in Scheduler, Add Support for Dynamic Reservations / Persistent Volumes

### ISSUE THREAD

**Issue by drewkerrigan** [mesos/mesos-go#187]
The protobuf messages for 0.24 have been added, but functionality is not yet implemented to utilize them.

Here is the documentation for persistent volumes: [http://mesos.apache.org/documentation/latest/persistent-volume/](http://mesos.apache.org/documentation/latest/persistent-volume/). From the docs: "`Offer::Operation::Create` and `Offer::Operation::Destroy` messages are available for frameworks to send back via the acceptOffers API as a response to a resource offer."

The `Create` and `Destroy` messages exist, but there is currently no way to call the `acceptOffers` function in the scheduler driver directly. 

Additionally, the use of persistent volumes requires that a resource be reserved using dynamic or static reservations. Support for dynamic reservations should also be added as a prerequisite to persistent volumes in the Go binding ([http://mesos.apache.org/documentation/latest/reservation/](http://mesos.apache.org/documentation/latest/reservation/)). From the docs: "`Offer::Operation::Reserve` and `Offer::Operation::Unreserve` messages are available for frameworks to send back via the acceptOffers API as a response to a resource offer."

Similarly, these messages exist, but cannot be used until support for direct incantation of the `acceptOffers` function within the scheduler driver is implemented.

Lastly, an example of using dynamic reservations and persistent volumes should be added to the `examples/` directory. For reference, here is the only existing example scheduler that I am aware of, it is written in C++: [https://github.com/apache/mesos/blob/master/src/examples/persistent_volume_framework.cpp](https://github.com/apache/mesos/blob/master/src/examples/persistent_volume_framework.cpp).

**Comment by drewkerrigan** [comment]
TL;DR: 
`scheduler.go` needs an `AcceptOffers` function or perhaps abstraction functions to `ReserveResrouce` / `UnreserveResource` (using `Offer::Operation::Reserve` and `Offer::Operation::Unreserve` messages) as well as `CreatePersistentVolume` / `DestroyPersistentVolume` (using `Offer::Operation::Create` and `Offer::Operation::Destroy` messages).

**Comment by tsenart** [comment]
@drewkerrigan: Thanks for filing. Please take a look at this: https://github.com/mesos/mesos-go/pull/188

**Comment by drewkerrigan** [comment]
@tsenart Thanks, I'm assuming the timing for the rewrite of the Go binding is dependent upon the release of the new HTTP API? Are you able to provide any rough estimates on when that will happen or how soon after the HTTP API is released the Go binding will be in feature parity?

**Comment by tsenart** [comment]
@drewkerrigan: The Mesos scheduler HTTP API is already out with the executor HTTP API coming in Mesos 0.26. We'll be working on the re-write starting this week but can't make any guarantees as to when it will be ready.

**Comment by drewkerrigan** [comment]
This was merged.
